moshi-adapters/src/test/kotlin/dev/zacsweers/moshix/adapters/JsonStringTest.kt

```diff
@@ -67,4 +74,39 @@ class JsonStringTest {
     val type: Int,
     @JsonString val rawJson: String?,
   )
+
+  @Test
+  fun retrofitServiceMethodCase() {
+    //language=JSON
+    val json = "{\"a\":2,\"b\":3,\"c\":[1,2,3]}"
+
+    val moshi = Builder()
+      .add(JsonString.Factory())
+      .build()
+
+    val aService = Retrofit.Builder()
+      .baseUrl("http://example.com/")
+      .callFactory(IdempotentCallFactory(json))
```

Comment:
Are you familiar with MockWebServer from OkHttp? I'd prefer to use that over the Callfactory approach here

Reply:
Yeah I can switch it to MockWebServer. Will update the PR in a bit.

Reply:
Done!